### PR TITLE
Bump versions for 0.5.0

### PR DESCRIPTION
**Pull request by joshmoore**
via

```
#!/usr/bin/env python

import hashlib

from fileinput import input
from glob import glob
from os.path import basename
from os.path import dirname
from os.path import exists
from os.path import join
from os import stat
from re import compile

DOWNLOADS = "/ome/www/downloads.openmicroscopy.org/"
URL_RE = compile("^(\s*RELEASE_URL\s+).*?$")
HASH_RE = compile("^(\s*RELEASE_HASH\s+\"SHA512=).*?$")

map = {
    "ome-common": "ome-common-cpp",
    "ome-files": "ome-files-cpp",
}

def sorted_ls(path):
    directories = glob("%s/[0-9]*" % path)
    timestamps = dict()
    for d in directories:
        timestamps[d] = stat(join(path, d)).st_mtime
    directories.sort(lambda a,b: cmp(timestamps[a], timestamps[b]))
    return directories

def sha512(fname):
    hash_sha512 = hashlib.sha512()
    with open(fname, "rb") as f:
        for chunk in iter(lambda: f.read(4096), b""):
            hash_sha512.update(chunk)
    return hash_sha512.hexdigest()

for x in glob("packages/ome*/superbuild.cmake"):
    dir = x.split("/")[1]
    dir = map.get(dir, dir)
    dir = "%s%s" % (DOWNLOADS, dir)
    if not exists(dir):
        print "Skip", dir
        continue

    versions = sorted_ls(dir)
    if not versions:
        print "No versions", dir
    else:
        print "Found", versions[-1]
        version = versions[-1].split("/")[-1]
        name = dir.split("/")[-1]
        source = "%s/%s/source/%s-%s.tar.xz" % (dir, version, name, version)
        hash = sha512(source)
        print "\t", source
        print "\t", sha512(source)
        for line in input([x], inplace=True):
            line = line.rstrip()
            line = URL_RE.sub(r'\g<1>"%s%s/%s/source/%s-%s.tar.xz"' % ("https://downloads.openmicroscopy.org/", name, version, name, version), line)
            line = HASH_RE.sub(r'\g<1>%s")' % hash, line)
            print line
```

### CODE REVIEW

**Comment by joshmoore** [comment]
```
jamoore@necromancer ~/ome-cmake-superbuild $ ./bump.py
Skip /ome/www/downloads.openmicroscopy.org/ome-files-performance
Found /ome/www/downloads.openmicroscopy.org/ome-common-cpp/5.5.0
	/ome/www/downloads.openmicroscopy.org/ome-common-cpp/5.5.0/source/ome-common-cpp-5.5.0.tar.xz
	9e82d245b73e5255fd0345f9bffafa3b8d512085c2ca26510418f325dccf497fd252a991c3b6204cdea8630f11b24697220ecd4f82c9f4649012c08fc7dddcd5
Skip /ome/www/downloads.openmicroscopy.org/ome-files-py
Skip /ome/www/downloads.openmicroscopy.org/ome-cmake-superbuild-docs-contents
Found /ome/www/downloads.openmicroscopy.org/ome-qtwidgets/5.4.3
	/ome/www/downloads.openmicroscopy.org/ome-qtwidgets/5.4.3/source/ome-qtwidgets-5.4.3.tar.xz
	00066fd21e7f2afe2762526977b307fb988747793e728c154556bee7ad340d8909c5be8c18de1b2ab8a3a2071182650ab70e6c1e2d4f15f148426ccc00716132
Found /ome/www/downloads.openmicroscopy.org/ome-files-cpp/0.5.0
	/ome/www/downloads.openmicroscopy.org/ome-files-cpp/0.5.0/source/ome-files-cpp-0.5.0.tar.xz
	682174514fc2ab9ab62a863d008a21ee50a64efa791c3016c2a9092b9f45fdd51eb667a8b9dd59c4cb58ce33fa64838ae9520a3ccb6d37d70500d81f69d9ce1a
Found /ome/www/downloads.openmicroscopy.org/ome-model/5.6.0
	/ome/www/downloads.openmicroscopy.org/ome-model/5.6.0/source/ome-model-5.6.0.tar.xz
	b19598df0c5b929d97707c809c25d139b24e105f8b4a17b1dfb04a34ea664215b5f659a1d5f09c6081a027f3c5230b7450549261d9d58162fade0c54146d105b
Skip /ome/www/downloads.openmicroscopy.org/ome-cmake-superbuild-docs
```

File: packages/ome-common/superbuild.cmake [hunk 1]
```diff
@@ -6,8 +6,8 @@ ome_source_settings(ome-common
   GIT_NAME        "ome-common-cpp"
   GIT_URL         "https://github.com/ome/ome-common-cpp.git"
   GIT_HEAD_BRANCH "master"
-  RELEASE_URL     "https://downloads.openmicroscopy.org/ome-common-cpp/5.4.2/source/ome-common-cpp-5.4.2.tar.xz"
-  RELEASE_HASH    "SHA512=db057145c5fe16d2664541789957d62609d4aa33705aab6719ef9731f0064a5911351fdce56e83d50c406c934e6688067634c8f9c96b395b13e2de4163785168")
+  RELEASE_URL     "https://downloads.openmicroscopy.org/ome-common-cpp/5.5.0/source/ome-common-cpp-5.5.0.tar.xz"
+  RELEASE_HASH    "SHA512=9e82d245b73e5255fd0345f9bffafa3b8d512085c2ca26510418f325dccf497fd252a991c3b6204cdea8630f11b24697220ecd4f82c9f4649012c08fc7dddcd5"
```
Comment (rleigh-codelibre) [inline packages/ome-common/superbuild.cmake]: Missing closing bracket.  This applies to all four changes.

Note, run `cmake ~/code/ome-cmake-superbuild` to quickly validate the correctness of these files.  I use `cmake -G Ninja -Dbuild-prerequisites=OFF ~/code/ome-cmake-superbuild` to verify the source downloading.

**Review by rleigh-codelibre — CHANGES_REQUESTED** [review]

File: packages/ome-common/superbuild.cmake (hunk 1 above)
Comment (joshmoore) [reply packages/ome-common/superbuild.cmake]: Pushed.

**Comment by rleigh-codelibre** [comment]
Successfully downloaded and built with:

- ome-common-cpp 5.5.0
- ome-model 5.6.0
- ome-files 0.5.0
- ome-qtwidgets 5.4.3

Looks good to merge to me.

**Comment by joshmoore** [comment]
Thanks, @rleigh-codelibre

**Pull request merged**